src/Director/BodyTo/BodyToProductPricing.php

```diff
@@ -13,7 +13,7 @@ public static function build(array $data): ProductPricing
     {
         return (new ProductPricing())
             ->setTaxIncluded($data['is_tax_included'])
-            ->setDiscountType($data['discount_type'])
+            ->setDiscountType($data['discount_type'] ?: null)
```

Comment:
Volgens mij is dit niet de oplossing. `public function setDiscountType(string $discountType): self` is niet nullable. $data['discount_type'] is al null.

Argument #1 ($discountType) must be of type string, **null given**